tests/core/pyspec/eth2spec/test/altair/transition/test_transition.py

```diff
@@ -242,3 +244,164 @@ def test_transition_only_blocks_post_fork(state, fork_epoch, spec, post_spec, pr
 
     yield "blocks", blocks
     yield "post", state
+
+
+@fork_transition_test(PHASE0, ALTAIR, fork_epoch=3)
+def test_transition_with_finality(state, fork_epoch, spec, post_spec, pre_tag, post_tag):
+    """
+    Transition from the initial ``state`` to the epoch after the ``fork_epoch``,
+    including attestations so as to produce finality through the fork boundary.
+    """
+    yield "pre", state
+
+    current_epoch = spec.get_current_epoch(state)
+    assert current_epoch < fork_epoch
+    assert current_epoch == spec.GENESIS_EPOCH
+
+    # regular state transition until fork:
+    fill_cur_epoch = False
+    fill_prev_epoch = True
+    blocks = []
+    for _ in range(current_epoch, fork_epoch - 1):
+        if current_epoch == spec.GENESIS_EPOCH:
+            fill_cur_epoch = True
```

Comment:
Isn't this going to fill both epoch 0 and epoch 1 with attestations from epoch 0? Not that bad, but just a funny consequence.

I might instead just do a `next_epoch_via_block` before the loop to get out of genesis epoch and then just do a prev_epoch fill from there with none of this exceptional logic to change and unchange these booleans